contrib/adaptive-compression/adapt.c

```diff
@@ -420,7 +422,7 @@ static void adaptCompressionLevel(adaptCCtx* ctx)
         /* use whichever one waited less because it was slower */
         double const completion = MAX(createWaitCompressionCompletion, writeWaitCompressionCompletion);
         unsigned const change = convertCompletionToChange(completion);
-        unsigned const boundChange = MIN(change, ctx->compressionLevel - 1);
+        unsigned const boundChange = ctx->compressionLevel >= g_minCLevel ? MIN(change, ctx->compressionLevel - g_minCLevel) : 0;
```

Comment:
You should check that the initial level is is in the range `[g_minCLevel, g_maxCLevel]`. Then you can skip the ternary expression in both cases, and maybe just add an assert that the compression level is in bounds.